static/js/stats/stats.js

```diff
@@ -55,6 +56,13 @@ function format_date(date, include_hour) {
     return month_str + ' ' + day + ', ' + year;
 }
 
+function final_data_end_times(last_full_update, data_end_times) {
```

Comment:
I'd call this `update_last_full_update`. I don't think it needs to take the first variable. I'd call the second variable `end_times`.